crates/nu-engine/src/env.rs

```diff
@@ -367,59 +367,48 @@ fn get_converted_value(
-                        let mut stack = stack.gather_captures(engine_state, &block.captures);
-                        if let Some(var_id) = &var.var_id {
-                            stack.add_var(*var_id, orig_val.clone());
-                        }
+    let conversions = stack.get_env_var(engine_state, ENV_CONVERSIONS);
+    let conversion = conversions
+        .as_ref()
+        .and_then(|val| val.as_record().ok())
+        .and_then(|record| record.get(name))
+        .and_then(|val| val.as_record().ok())
+        .and_then(|record| record.get(direction));
+
+    if let Some(conversion) = conversion {
+        let from_span = conversion.span();
+        match conversion.as_closure() {
+            Ok(val) => {
+                let block = engine_state.get_block(val.block_id);
+
+                if let Some(var) = block.signature.get_positional(0) {
+                    let mut stack = stack.gather_captures(engine_state, &block.captures);
+                    if let Some(var_id) = &var.var_id {
+                        stack.add_var(*var_id, orig_val.clone());
+                    }
 
-                        let result = eval_block(
-                            engine_state,
-                            &mut stack,
-                            block,
-                            PipelineData::new_with_metadata(None, val_span),
-                            true,
-                            true,
-                        );
-
-                        match result {
-                            Ok(data) => ConversionResult::Ok(data.into_value(val_span)),
-                            Err(e) => ConversionResult::ConversionError(e),
-                        }
-                    } else {
-                        ConversionResult::ConversionError(ShellError::MissingParameter {
-                            param_name: "block input".into(),
-                            span: from_span,
-                        })
+                    let val_span = orig_val.span();
+                    let result = eval_block(
+                        engine_state,
+                        &mut stack,
+                        block,
+                        PipelineData::new_with_metadata(None, val_span),
+                        true,
+                        true,
+                    );
+
+                    match result {
+                        Ok(data) => ConversionResult::Ok(data.into_value(val_span)),
+                        Err(e) => ConversionResult::ConversionError(e),
                     }
+                } else {
+                    ConversionResult::ConversionError(ShellError::MissingParameter {
+                        param_name: "block input".into(),
+                        span: from_span,
+                    })
                 }
-                _ => ConversionResult::CellPathError,
             }
-        } else {
-            ConversionResult::CellPathError
+            _ => ConversionResult::CellPathError,
```

Comment:
I think this should instead propagate the `ShellError` as `ConversionResult::ConversionError`. Also, maybe the `ShellError::CantConvert` should be transformed to `ShellError::UnsupportedConfigValue`.

Reply:
Yeah feel free to map that to more appropriate errors as you see fit.